### PR TITLE
Fix global CLI options

### PR DESCRIPTION
**Pull request by dnewh**
This moves the key and service-id args one level deeper. These args were previously global to all grid commands but were not relevant to keygen and admin subcommands. Now these args must be specified after the grid subcommand rather than the grid command and are only available on relevant subcommands. This also updates the Grid-on-Splinter example README to reflect these changes.

### CODE REVIEW

**Review by dplumb94 — CHANGES_REQUESTED** [review]
This should include the --wait to only show for creation commands and --url only for commands that have to do with the REST API.

**Review by dplumb94 — DISMISSED** [review]

File: cli/src/main.rs [hunk 1]
```diff
@@ -178,25 +191,32 @@ fn run() -> Result<(), CliError> {
         (@subcommand product =>
             (about: "Create, update, or delete products")
             (@setting SubcommandRequiredElseHelp)
+            (@arg url: --url +takes_value "URL for the REST API")
+            (@arg key: -k +takes_value "base name for private key file")
```
Comment (peterschwarz) [inline cli/src/main.rs:195]: Isn't the key command only relevant to the commands that also have `wait`? I.e. both are related to submitting transactions.

Comment (dnewh) [reply cli/src/main.rs:195]: I believe you also need it for fetching records unlike wait

Comment (peterschwarz) [reply cli/src/main.rs:195]: That does not look to be the case

Comment (dnewh) [reply cli/src/main.rs]: updated

**Review by peterschwarz — APPROVED** [review]

**Review by dplumb94 — APPROVED** [review]

**Pull request merged**